global-script.lua

```diff
@@ -3245,4 +3410,282 @@ function tCompare(t1,t2)
+end
+
+function swapPlayerAreaColors(a, b)
+    if a == b then return end
+    local function tableSwap(table)
+        table[a], table[b] = table[b], table[a]
+    end
+    local function tintSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        local ta = oa.getColorTint()
+        local tb = ob.getColorTint()
+        oa.setColorTint(tb)
+        ob.setColorTint(ta)
+        tableSwap(table)
+    end
+    local function positionSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getPosition()
+        local tb = ob.getPosition()
+        oa.setPosition(tb)
+        ob.setPosition(ta)
+    end
+    local function handSwap(i)
+        local ta = Player[a].getHandTransform(i)
+        local tb = Player[b].getHandTransform(i)
+        Player[a].setHandTransform(tb, i)
+        Player[b].setHandTransform(ta, i)
+    end
+    local function updateBlock(color)
+        local o = playerBlocks[color]
+        if not o then
+            return 
+        end
+        o.setVar("playerColor", color)
+        o.editButton({index=1, label="Swap with " .. color})
+    end
+
+    for i = 1,2 do
+        handSwap(i)
+    end
+    positionSwap(playerTables)
+    tableSwap(playerBlocks)
+    positionSwap(defendBags)
+    tableSwap(elementScanZones)
+    updatePlayerArea(a)
+    updatePlayerArea(b)
+end
+
+function swapPlayerAreaObjects(a, b) 
+    if a == b then return end
+    local swaps = {[a] = b, [b] = a}
+    local tables = {[a] = playerTables[a], [b] = playerTables[b]}
+    local zones = {}
+    local buttons = {}
+    local objects = {}
+    for color,playerTable in pairs(tables) do
+        local t = upCast(playerTable, 50)
+        for _,obj in ipairs(Player[color].getHandObjects(2)) do
+            table.insert(t, obj)
+        end
+        objects[color] = t
+        local zone = getObjectFromGUID(elementScanZones[color])
+        zones[color] = zone
+        local zoneButtons = zone.getButtons()
+        buttons[color] = zoneButtons
+        if zoneButtons then
+            for i = #zoneButtons - 1, 0, -1 do
+                zone.removeButton(i)
+            end
+        end
+    end
+    for from,to in pairs(swaps) do
+        local transform = tables[to].getPosition() - tables[from].getPosition()
+        for _,obj in ipairs(objects[from]) do
+            if obj.interactable then
+                obj.setPosition(obj.getPosition() + transform)
+            end
+        end
+        if buttons[from] then
+            for _,button in ipairs(buttons[from]) do
+                zones[to].createButton(button)
+            end
+        end
+    end
+end
+
+function swapPlayerAreas(a, b)
+    if(a == b) then return end
+    swapPlayerAreaObjects(a, b)
+    swapPlayerAreaColors(a, b)
+    print(a .. " swapped places with " .. b .. ".")
```

Comment:
this should probably be changed to broadcast to be consistent with the mod

Reply:
broadcastToAll() might be a little obnoxious with a bunch of broadcast messaging as players claim their initial starts as a game starts, but it's now changed to printToAll() and color-coded to the initiating player.